feature-libs/product-configurator/rulebased/cpq/cpq-configurator-utils.service.ts

```diff
@@ -0,0 +1,298 @@
+import { Injectable, isDevMode } from '@angular/core';
+import { getLocaleId } from '@angular/common';
+import { Configurator } from '../core/model/configurator.model';
+import { Cpq } from './cpq.models';
+import { formatCurrency, getCurrencySymbol } from '@angular/common';
+import { LanguageService } from '@spartacus/core';
+
+/**
+ * Utilities for CPQ configuration
+ */
+@Injectable({ providedIn: 'root' })
+export class CpqConfiguratorUtilsService {
```

Comment:
This name seems too general. What about `CpqConfiguratorNormalizerUtils`? If I get correctly, it's job is to normalize properties of the CPQ configuration into the generic Configuration model. Btw. for consistency, I'm in favor of harmonizing the public methods' names: `convertX` instead of `prepareX`, `calculateX`, `retrieveX`.

Reply:
Picked

Reply:
The util class is renamed into CpqConfiguratorNormalizerUtilsService as discussed. Methods are renamed as suggested. Two methods will remain calculateXXX, since they are really calculate something, eg calculate total price base on the quantity and item price.